invisible_cities/reco/icaro_components_test.py

```diff
@@ -0,0 +1,136 @@
+    pdt.assert_series_equal(y_data, expected_y)
+
+
+def test_prepare_data_expo(sample_dataframe):
+
+    x_data, y_data = icarcomp.prepare_data(KrFitFunction.expo, sample_dataframe)
+
+    expected_x = sample_dataframe['DT']
+    expected_y = sample_dataframe['S2e']
+
+    pdt.assert_series_equal(x_data, expected_x)
+    pdt.assert_series_equal(y_data, expected_y)
+
+
+def test_prepare_data_log_lin(sample_dataframe):
+
+    x_data, y_data = icarcomp.prepare_data(KrFitFunction.log_lin, sample_dataframe)
+
+    expected_x = sample_dataframe['DT']
+    expected_y = -np.log(sample_dataframe['S2e'])
+
+    pdt.assert_series_equal(x_data, expected_x)
+    pdt.assert_series_equal(y_data, expected_y)
+
+
+def test_get_fit_function_lt():
+
+    expected_functions = {
+        KrFitFunction.linear: (lambda x, a, b: a + b * x, icarcomp.lin_seed),
+        KrFitFunction.expo: (lambda x, const, mean: const * np.exp(-x / mean), icarcomp.expo_seed),
+        KrFitFunction.log_lin: (lambda x, a, b: a + b * x, icarcomp.lin_seed)
+    }
+
+    for fit_type, (expected_fit_function, expected_seed_function) in expected_functions.items():
+        fit_function, seed_function = icarcomp.get_fit_function_lt(fit_type)
+
+        assert fit_function.__name__  == expected_fit_function.__name__
+        assert seed_function.__name__ == expected_seed_function.__name__
+
+
+@pytest.mark.parametrize("fittype", [KrFitFunction.linear, KrFitFunction.expo, KrFitFunction.log_lin])
+@given(floats(min_value=1,  max_value=1e5),
+       floats(min_value=10, max_value=1e5))
+def test_transform_parameters(fittype, a, b):
+
+    errors = 0.2*np.array([a, b])
+
+    if fittype == KrFitFunction.log_lin:
+
+        fit_output = FitFunction(values=[a, b], errors=errors, cov=np.array([[0.04, 0.02], [0.02, 0.04]]),
+                                 fn=None, chi2=None, pvalue=None)
+
+        transformed_par, transformed_err, transformed_cov = icarcomp.transform_parameters(fittype, fit_output)
+
+        E0_expected   = np.exp(-a)
+        s_E0_expected = np.abs(E0_expected * errors[0])
+        lt_expected   = 1 / b
+        s_lt_expected = np.abs(lt_expected**2 * errors[1])
+        cov_expected  = E0_expected * lt_expected**2 * 0.02
+
+
+        npt.assert_allclose(transformed_par, [  E0_expected,   lt_expected], rtol=0.01)
+        npt.assert_allclose(transformed_err, [s_E0_expected, s_lt_expected], rtol=0.01)
+        assert np.isclose(transformed_cov, cov_expected, rtol=0.01)
+
+    else:
+
+        fit_output = FitFunction(values=[a, b], errors=errors, cov=np.array([[0.04, 0.02], [0.02, 0.04]]),
+                                 fn=None, chi2=None, pvalue=None)
+
+        transformed_par, transformed_err, transformed_cov = icarcomp.transform_parameters(fittype, fit_output)
+
+        npt.assert_allclose(transformed_par, [a, b], rtol=0.01)
+        npt.assert_allclose(transformed_err, errors, rtol=0.01)
+
+        assert np.isclose(transformed_cov, 0.02, rtol=0.01)
```

Comment:
If you rewrite the function to consider only the log_lin case, it will simplify the test. Again, the test is just a repetition of the implementation. Perhaps we can check some properties, rather than the exact values